src/H5.c

```diff
@@ -332,8 +333,13 @@ H5_term_library(void)
         while (curr_atclose) {
             H5_atclose_node_t *tmp_atclose; /* Temporary pointer to 'atclose' node */
 
-            /* Invoke callback, providing context */
-            (*curr_atclose->func)(curr_atclose->ctx);
+            /* Prepare & restore library for user callback */
```

Comment:
Example of macro wrappers around callback invocations that could call an application function, and therefore re-enter the library.